src/common/HeartbeatMap.cc

```diff
@@ -78,7 +79,10 @@ bool HeartbeatMap::_check(const heartbeat_handle_d *h, const char *who, time_t n
   if (was && was < now) {
     ldout(m_cct, 1) << who << " '" << h->name << "'"
 		    << " had suicide timed out after " << h->suicide_grace << dendl;
-    assert(0 == "hit suicide timeout");
+    if (h->thread_id == 0)
+      assert(0 == "hit suicide timeout");
+    else
+      pthread_kill(h->thread_id, SIGABRT);
```

Comment:
Does this trigger a stack trace dump on the other thread?  SIGABRT or something?